cmake/modules/Findc-ares.cmake

```diff
@@ -28,3 +28,15 @@ if(c-ares_FOUND AND NOT (TARGET c-ares::c-ares))
     IMPORTED_LINK_INTERFACE_LANGUAGES "C"
     IMPORTED_LOCATION "${c-ares_LIBRARY}")
 endif()
+
+# Although it's essentially a duplicate of the above compound
+# statement, this is necessary because gRPC::grpc++ refers to
+# c-ares::cares instead of c-ares::c-ares.
+if(c-ares_FOUND AND NOT (TARGET c-ares::cares))
+  add_library(c-ares::cares ALIAS c-ares::c-ares)
+  # add_library(c-ares::cares UNKNOWN IMPORTED)
+  # set_target_properties(c-ares::cares PROPERTIES
+  # INTERFACE_INCLUDE_DIRECTORIES "${c-ares_INCLUDE_DIR}"
+  # IMPORTED_LINK_INTERFACE_LANGUAGES "C"
+  # IMPORTED_LOCATION "${c-ares_LIBRARY}")
+endif()
```

Comment:
instead of repeating and chaining the targets, i still think https://github.com/ceph/ceph/pull/46121#discussion_r863816870 is a better change.

Reply:
I did incorporate that full change, @tchaikov . However ci's `make check` is now giving me an error:
```
CMake Error at cmake/modules/Findc-ares.cmake:33 (add_library):
  add_library cannot create ALIAS target "c-ares::c-ares" because target
  "c-ares" is imported but not globally visible.
```

I do not know if the previous version produced that same error in ci's `make check`. I see that there's a `GLOBAL` option for `add_library`, but that was never there in your prior version of `Findc-ares.cmake`.

Reply:
ping @tchaikov

Reply:
@ivancich sorry for the latency. yeah, see we need to add `GLOBAL` in order to create `ALIAS` to it. another way to achieve similar goal is to create `INTERFACE` library to `c-ares`, but that'd take two commands:
```cmake
add_library(c-ares::c-ares INTERFACE)
target_link_libraries(c-ares::c-ares c-ares)
```

so i took the liberty to add `GLOBAL` to `Findc-ares.cmake` in your changeset.